index.js

```diff
@@ -21,38 +32,35 @@ const uuid = require('uuid/v1');
  * amqp.routingKey - the RK to publish logs to
  */
 function main (config) {
-  const connected = Promise.resolve().then(function () {
-    return AMQP(config.amqp);
-  }).then(function (amqp) {
-    return amqp.connect().then(function () { return amqp; });
-  }).catch(function (err) {
-    debug('Error connecting to amqp', err);
-    return 'failed';
-  });
+  const _amqp = getAmqp(config);
 
   return function getLogger () {
     let flushed = false;
     let logs = [];
 
-    function flush () {
+    async function flush (opts = {}) {
+      const { meta } = opts;
       if (flushed) throw new Error('Already flushed.');
-      flushed = true;
-      return connected.then(function (amqp) {
-        if (amqp === 'failed') return Promise.resolve();
-        return amqp.publish(config.amqp.routingKey, {
+      try {
+        flushed = true;
+        const amqp = await _amqp;
+        if (amqp === 'failed') return;
+        const payload = {
           id: uuid(),
           timestamp: Date.now(),
-          'log2amqp-schema-version': '2.0.0',
+          'log2amqp-schema-version': '2.1.0',
           source: config.source,
-          logs })
-          .catch(console.error)
-          .then(() => {
-            // We explicitly set this to undefined just in case somehow there is a
-            // reference back to the logger in the logged payload.
-            // Such a circular reference would prevent garbage collection.
-            logs = undefined;
-          });
-      });
+          logs };
+        if (meta) payload.meta = meta;
+        await amqp.publish(config.amqp.routingKey, payload);
```

Comment:
would it be worth checking if `logs` actually contains anything in here somewhere before we publish it?

Reply:
Possibly, though perhaps we can leave that up to the user to not call `flush()` if they never called `log()`...?